nixos/modules/misc/nixpkgs.nix

```diff
@@ -157,6 +175,46 @@ in
       '';
     };
 
+    hostSystem = mkOption {
+      type = types.either types.str types.attrs; # TODO utilize lib.systems.parsedPlatform
+      example = { system = "aarch64-linux"; config = "aarch64-unknown-linux-gnu"; };
+      # Make sure that the final value has all fields for sake of other modules
+      # referring to this. TODO make `lib.systems` itself use the module system.
+      apply = lib.systems.elaborate;
+      defaultText = literalExpression
+        ''(import "''${nixos}/../lib").lib.systems.examples.aarch64-multiplatform'';
+      description = ''
+        Specifies the platform where the NixOS configuration will run.
+
+        To cross-compile, set also <code>nixpkgs.buildSystem</code>.
+
+        Ignored when <code>nixpkgs.pkgs</code> is set.
+      '';
+    };
+
+    buildSystem = mkOption {
+      type = types.either types.str types.attrs; # TODO utilize lib.systems.parsedPlatform
+      default = cfg.hostSystem;
+      example = "x86_64-linux";
```

Comment:
```suggestion
      example = { system = "aarch64-linux"; config = "aarch64-unknown-linux-gnu"; };
```

Reply:
Done, but I picked `x86_64` because it's more "dev machine"-like, at least for now.